src/detect-engine-analyzer.c

```diff
@@ -1047,6 +1047,8 @@ void EngineAnalysisRules2(const DetectEngineCtx *de_ctx, const Signature *s)
             break;
     }
 
+    jb_set_bool(ctx.js, "rule_state_dependency", s->init_data->rule_state_dependency);
```

Comment:
I think the immediate question ppl will have is "which rule?" followed by "why?". Perhaps more info can be added through the analyzer note part of the engine analysis